map-feature bug fixes

AliyanH opened the pull request:
- [X] #838 
- [X] make click/focus/blur events work and in sync with the g element (can work as attribute, API, and addEventListener).
- [X] Allow prevention of default click handling, can be done as following:
```js
// $0 represents the map-feature element
$0.addEventListener("click", (e) => {
   e.originalEvent.stopPropagation();
});
```
- [X] add support for keyup/keydown event on map-feature
- [x] Add Tests

closes #838

prushforth commented:
I would have liked to get rid of the click(ev), focus(ev) and blur(ev) parameter, as it seems [non-standard](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click), but perhaps that's not possible because we have got leaflet handling the underlying clicks and displaying popups and so on.

prushforth commented on src/map-feature.js:
I think that you removed the onfocus, onclick and onblur attributes, so no need to watch them here.

prushforth replied:
... although we could reasonably add min and max to the observedAttributes array https://maps4html.org/web-map-doc/docs/api/map-feature-api#min

AliyanH commented on src/map-feature.js:
```diff
@@ -578,34 +560,34 @@ export class MapFeature extends HTMLElement {
       }
+      if (featureGroup.isPopupOpen()) {
+        featureGroup.closePopup();
+      } else if (!clickEv.originalEvent.cancelBubble) {
```
`cancelable` is only used to check if an event can be cancelled. cancelBubble is true when when the user stops Propagation.

AliyanH commented:
noticed another issue just now that I will work on, removing a map-feature when it is unchecked does not work.

AliyanH commented:
I think this is good to go @prushforth , just a few changes I added: 
- removed non-standard parameters for click, focus, blur methods (based on suggestion by @prushforth)
- Added support for keyup/keydown on map-feature

prushforth commented:
Would you mind fixing [this bug](https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/855) in this PR?  If that doesn't make sense, feel free to merge this, it looks and works well.

prushforth commented:
I like the test for preventDefault that displays a different thing than the popup for the polygon, good idea there.  It tests the API and not just our API; that's exactly what we need to do more of.
